consensus/consensus-types/src/vote.rs

```diff
@@ -162,4 +160,20 @@ impl Vote {
         self.vote_data().verify()?;
         Ok(())
     }
+
+    /// Verifies the signature on the LedgerInfo.
+    pub fn verify_signature(&self, validator: &ValidatorVerifier) -> anyhow::Result<()> {
+        // TODO: Should we verify only if the status is unverified?
```

Comment:
either we have verified signatures don't come here, or we can remove the verified/unverified signatures for the LI (since verify becomes trivial)

Reply:
I actually didn't understand this -- "we can remove the verified/unverified signatures for the LI". You mean to remove the status information?

Reply:
no, I mean on the LedgerInfoWithUnverifiedSignatures we don't need to maintain two PartialSignatures, since signatures carry the status inside and verify a verified signature is trivial

Reply:
I don't have a strong opinion here. Don't you think the code looks cleaner and more readable when the verified and unverified signatures separated out? If you prefer to merge them, I can do that too.

Reply:
it doesn't matter much, i was just saying if we don't go that route, here we should expect the signature is unverified and not verify a verified signature again